Gems/Atom/RPI/Code/Source/RPI.Public/Material/Material.cpp

```diff
@@ -60,18 +60,19 @@ namespace AZ
             ScopedValue isInitializing(&m_isInitializing, true, false);
 
             // All of these members must be reset if the material can be reinitialized because of the shader reload notification bus
+            m_materialAsset = { &materialAsset, AZ::Data::AssetLoadBehavior::PreLoad };
             m_shaderResourceGroup = {};
             m_rhiShaderResourceGroup = {};
             m_materialProperties = {};
             m_generalShaderCollection = {};
             m_materialPipelineData = {};
-            m_materialAsset = { &materialAsset, AZ::Data::AssetLoadBehavior::PreLoad };
+
             ShaderReloadNotificationBus::MultiHandler::BusDisconnect();
             if (!m_materialAsset.IsReady())
             {
-                // We will call this function again later when the asset is ready.
-                Data::AssetBus::Handler::BusConnect(m_materialAsset.GetId());
-                return RHI::ResultCode::Success;
+                AZ_Error(s_debugTraceName, false, "Material::Init failed because asset is not ready. materialAsset uuid=%s",
+                    materialAsset.GetId().ToFixedString().c_str());
+                return RHI::ResultCode::Fail;
```

Comment:
I don't see anything in this PR that changes the material loading and initializing flow, why doesn't this need to listen to the AssetBus any more and can assume that it's always loaded by the time Init() is called?

Reply:
Fair point. The last few PRs were broken out of a larger PR to make them more digestible. This file did not get divided into separate pull requests. The material must be ready before Init is called. Returning success when the material is not initialized and relying on it to reload itself is incorrect. I will search for the PR where that was introduced.